money.go

```diff
@@ -4,28 +4,59 @@ import (
 	"errors"
 )
 
-type amount struct {
-	val int
+// Amount is a datastructure that stores the amount being used for calculations
+type Amount struct {
+	val int64
 }
 
 // Money represents monetary value information, stores
 // currency and amount value
 type Money struct {
-	amount   *amount
+	amount   *Amount
 	currency *Currency
 }
 
 var calc *calculator
```

Comment:
Yea, that's a good point. Must be removed and changed.
What do you think about init() all manipulators (Calculator, Formatter, as I showed you in graph) in one struct?